docs/antrea-l7-network-policy.md

```diff
@@ -0,0 +1,204 @@
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: NetworkPolicy
+metadata:
+  name: ingress-allow-http-request-to-api-v2
+spec:
+  priority: 5
+  tier: application
+  appliedTo:
+    - podSelector:
+        matchLabels:
+          app: web
+  ingress:
+    - name: allow-http   # Allow inbound HTTP GET requests to "/api/v2" from Pods with app=client label.
+      action: Allow      # All other traffic from these Pods will be automatically dropped, and subsequent rules will not be considered.
+      from:
+        - podSelector:
+            matchLabels:
+              app: client
+      l7Protocols:
+        - http:
+            path: "/api/v2"
+            host: "foo.bar.com"
+            method: "GET"
+    - name: drop-other   # Drop all other inbound traffic (i.e., from Pods without the app=client label or from external clients).
+      action: Drop
+```
+
+**path**: The `path` field represents the URI path to match (e.g. "/api/v2", "/admin"). If not set, the rule matches all
+URI paths.
+
+**host**: The `host` field represents the hostname present in the URI or the HTTP Host header to match. It does not
+contain the port associated with the host. If not set, the rule matches all hostnames.
+
+**method**: The `method` field represents the HTTP method to match. It could be GET, POST, PUT, HEAD, DELETE, TRACE,
+OPTIONS, CONNECT and PATCH. If not set, the rule matches all methods.
+
+#### More examples
+
+The following NetworkPolicy grants access of privileged URLs to specific clients while make other URLs publicly
+accessible:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: NetworkPolicy
+metadata:
+  name: allow-privileged-url-to-admin-role
+spec:
+  priority: 5
+  tier: securityops
+  appliedTo:
+  - podSelector:
+      matchLabels:
+        app: web
+  ingress:
+  - name: for-admin    # Allow inbound HTTP GET requests to "/admin" and "/public" from Pods with role=admin label.
+    action: Allow
+    from:
+    - podSelector:
+        matchLabels:
+          role: admin
+    l7Protocols:
+    - http:
+        path: "/admin/"
+    - http:
+        path: "/public/"
+  - name: for-public   # Allow inbound HTTP GET requests to "/public" from Pods with app=client label.
+    action: Allow      # All other inbound traffic will be automatically dropped.
+    l7Protocols:
+      - http:
+          path: "/public/"
+```
+
+The following NetworkPolicy prevents applications from accessing unauthorized domains:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: allow-web-access-to-internal-domain
+spec:
+  priority: 5
+  tier: securityops
+  appliedTo:
+  - podSelector:
+      matchLabels:
+        egress-restriction: internal-domain-only
+  egress:
+  - name: allow-dns    # Allow outbound DNS requests.
+    action: Allow
+    ports:
+      - protocol: TCP
+        port: 53
+      - protocol: UDP
+        port: 53
+  - name: allow-http   # Allow outbound HTTP requests towards foo.bar.com.   
```

Comment:
maybe let's rename the rule to `allow-http-only`

Reply:
done